src/coreComponents/common/MpiWrapper.hpp

```diff
@@ -863,15 +854,16 @@ int MpiWrapper::iRecv( T * const buf,
                        MPI_Request * MPI_PARAM( request ) )
 {
 #ifdef GEOSX_USE_MPI
+  GEOSX_ERROR_IF( (*request)!=MPI_REQUEST_NULL && (*request)!=0,
```

Comment:
What does the comparison of an `MPI_Request` handle with `0` do? I think that happens to work in both OpenMPI and MPICH due to chosen representations (pointer and int), but not guaranteed to be valid in general, since handle types are not standardized?

Reply:
Yeah per the standard I think all we can guarantee is an `MPI_Request` handle is assignable the `MPI_REQUEST_NULL` value and comparable against that value, not that `0` is coercible to a comparable type / value.

Reply:
The intent is to check whether or not the `MPI_Request` is in use. i.e. the `MPI_Request` has been used in a non-blocking operation, but has not been "cleared" by an `MPI_Wait` etc. The completion of `MPI_Wait` means  `MPI_Request==MPI_REQUEST_NULL`. The `(*request)!=0` covers the case where the value has never been used.

Reply:
oh...is `MPI_Request` an `int` in the standard?

Reply:
No per the standard it is an opaque handle.

Reply:
So you can compare it to other `MPI_Request`s and `MPI_REQUEST_NULL`, but comparing it against anything else isn't necessarily well-defined. MPICH has it as an `int`, openmpi has it as a pointer, in general it could be anything.

Reply:
> So you can compare it to other `MPI_Request`s and `MPI_REQUEST_NULL`, but comparing it against anything else isn't necessarily well-defined. MPICH has it as an `int`, openmpi has it as a pointer, in general it could be anything.

im lazy

Reply:
OK. I removed that `0` check, and initialized all `MPI_Request` to `MPI_REQUEST_NULL`. It is a little heavy handed, but it is less error prone....but more incorrect usage prone.